scrapy/extensions/feedexport.py

```diff
@@ -93,12 +93,28 @@ def store(self, file):
 
 class S3FeedStorage(BlockingFeedStorage):
 
-    def __init__(self, uri):
+    def __init__(self, uri, access_key=None, secret_key=None):
+        # BEGIN Backwards compatibility for initialising without keys (and
+        # without using from_crawler)
         from scrapy.conf import settings
```

Comment:
it can be a good time to do this deprecated import only if `no_defaults` is True